taxonomy/models.py

```diff
@@ -102,6 +102,125 @@ class Meta:
         app_label = 'taxonomy'
 
 
+class XBlockSkills(TimeStampedModel):
+    """
+    Skills that will be learnt by completing xblock.
+
+    .. no_pii:
+    """
+
+    usage_key = models.CharField(
+        unique=True,
+        max_length=255,
+        help_text=_('The key of the xblock whose text was used for skills extraction.')
+    )
+    skills = models.ManyToManyField(
+        Skill,
+        through="XBlockSkillThrough",
+        help_text=_(
+            'The ID of the skill extracted for the xblock.'
+        )
+    )
+    requires_verification = models.BooleanField(
+        default=True,
+        help_text=_('Indicates whether skills applied to this block requires verification from users'),
+    )
+    auto_processed = models.BooleanField(
+        default=False,
+        help_text=_('Indicates whether the text from this block was already processed'),
+    )
+    hash_content = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        help_text=_('Hashed text content useful for checking if content has changed')
+    )
+
+    class Meta:
+        """
+        Meta configuration for CourseSkills model.
+        """
+
+        verbose_name = 'XBlock Skills'
+        verbose_name_plural = 'XBlock Skills'
+        ordering = ('created', )
+        app_label = 'taxonomy'
+
+    def __str__(self):
+        """
+        Create a human-readable string representation of the object.
+        """
+        return '<XBlockSkills usage_key="{}">'.format(self.usage_key)
+
+    def __repr__(self):
+        """
+        Create a unique string representation of the object.
+        """
+        return '<XBlockSkills id="{0}">'.format(self.id)
+
+
+class XBlockSkillThrough(TimeStampedModel):
```

Comment:
```suggestion
class XBlockSkillsData(TimeStampedModel):
```

Through is very djangoy term, this is an entity on it's own which describes the relationship between an xblock and a skill and has various data point/information about it. What do you think?